tools/figma-inspector/src/utils/store.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+import { create } from "zustand";
+import { dispatchTS, listenTS } from "./bolt-utils";
+import { downloadZipFile, writeTextToClipboard } from "./utils.js";
+
+interface StoreState {
+    title: string;
+    slintSnippet: string;
+    useVariables: boolean;
+    exportsAreCurrent: boolean;
+    exportedFiles: Array<{ name: string; content: string }>;
+    exportAsSingleFile: boolean;
+    menuOpen: boolean;
+    setTitle: (title: string) => void;
+    initializeEventListeners: () => void;
+    copyToClipboard: () => Promise<void>;
+    setUseVariables: (useVariables: boolean) => void;
+    setExportsAreCurrent: (exportsAreCurrent: boolean) => void;
+    setExportAsSingleFile: (exportAsSingleFile: boolean) => void;
+    exportFilesHandler: (
+        files: Array<{ name: string; content: string }>,
+    ) => Promise<void>;
+    setMenuOpen: (menuOpen: boolean) => void;
+    toggleMenu: () => void;
+    exportFiles: () => void;
+}
+
+export const useInspectorStore = create<StoreState>()((set, get) => ({
+    // Default store values
+    title: "",
+    slintSnippet: "",
+    useVariables: false,
+    exportsAreCurrent: false,
+    exportedFiles: [],
+    exportAsSingleFile: false,
+    menuOpen: false,
+
+    setTitle: (title) => set({ title }),
+
+    initializeEventListeners: () => {
+        listenTS("updatePropertiesCallback", (res) => {
+            set({ title: res.title, slintSnippet: res.slintSnippet || "" });
+        });
+
+        listenTS("selectionChangedInFigma", () => {
+            dispatchTS("generateSnippetRequest", {
+                useVariables: get().useVariables,
+            });
+        });
+
+        listenTS("exportedFiles", (res) => {
+            get().exportFilesHandler(res.files);
+        });
+    },
+
+    copyToClipboard: async () => {
+        try {
+            writeTextToClipboard(get().slintSnippet);
```

Comment:
Otherwise the `dispatchTS("copyToClipboard", { result: true });` is *always* called, because invoking the function without `await` will just spawn the promise.

```suggestion
            await writeTextToClipboard(get().slintSnippet);
```

Reply:
I'm going to look at why the lint rule for dangling async functions isn't enabled to stop this kind of thing in the future.